Adds new section that better explains ESM context and TypeScript exports (#2484)
fastify/fastify: docs/Plugins-Guide.md

export default plugin
```

> __Note__: Fastify doesn't support named imports in ESM context. However, it is not an issue at all, since you can always use default import. 

Reviewer: ```suggestion
__Note__: Fastify doesn't support named imports within an ESM context. Instead, the `default` export is available. 
```
